mempool/src/shared_mempool/priority.rs

```diff
@@ -208,11 +242,149 @@ impl PrioritizedPeersState {
+        // TODO: (Minor) If the load is low, don't do load balancing for Failover buckets.
+        assert!(self.prioritized_peers.read().len() == peer_monitoring_data.len());
+
+        // Obtain the top peers to assign the sender buckets with Primary priority
+        let mut top_peers = vec![];
+        let time_elapsed_since_last_update =
+            self.last_peer_priority_update.map_or(0, |last_update| {
+                self.time_service
+                    .now()
+                    .duration_since(last_update)
+                    .as_secs()
+            });
+
+        // If the observed traffic is to be load balanced across upstream peers such that each upstream peer
+        // receives `avg_mempool_traffic_in_tps_per_peer` transactions per second, then this is the number of
+        // upstream peers required.
+        // This is calculated so that when the traffic is low, we don't do load balancing across many peers.
+        let num_peers_required_for_load_balancing = min(
+            (num_txns_received_since_peers_updated as f64
+                / ((self.mempool_config.avg_mempool_traffic_in_tps_per_peer
+                    * max(1, time_elapsed_since_last_update)) as f64))
+                .ceil() as u64,
+            MempoolSenderBucket::MAX.into(),
+        ) as MempoolSenderBucket;
+        let num_top_peers = max(
+            1,
+            min(
+                self.mempool_config.num_sender_buckets,
+                num_peers_required_for_load_balancing,
+            ),
+        );
+        info!(
+            "Time elapsed since last peer update: {:?}\n
+               Number of transactions received since last peer update: {:?},\n
+               Ideal number of peers required for load balancing: {:?},\n
+               Number of top peers picked: {:?}",
+            time_elapsed_since_last_update,
+            num_txns_received_since_peers_updated,
+            num_peers_required_for_load_balancing,
+            num_top_peers
+        );
+
+        if self.node_type.is_validator_fullnode() {
+            // Use the peer on the VFN network with lowest ping latency as the primary peer
+            let peers_in_vfn_network = self
+                .prioritized_peers
+                .read()
+                .iter()
+                .cloned()
+                .filter(|peer| peer.network_id() == NetworkId::Vfn)
+                .collect::<Vec<_>>();
+
+            if !peers_in_vfn_network.is_empty() {
+                top_peers = vec![peers_in_vfn_network[0]];
+            }
+        }
+
+        if top_peers.is_empty() {
+            let base_ping_latency = self.prioritized_peers.read().first().and_then(|peer| {
+                peer_monitoring_data
+                    .get(peer)
+                    .and_then(|metadata| get_peer_ping_latency(metadata))
+            });
+
+            // Extract top peers with ping latency less than base_ping_latency + 30 ms
```

Comment:
change 30ms to latency_slack_between_top_upstream_peers in the comment

Reply:
Done.